### PR TITLE
Improve Document's Path API

### PR DESCRIPTION
**Pull request by w84thesun**
# Description

This PR refs #803.

In that PR I added the set of `Document` methods working with `Path`. They would be needed later to implement dot notation support for update operators.

Here we came up with the decision that a `Path` with one element is the correct path and it's a caller's responsibility to check where that case is applicable.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers, Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by rumyantseva — APPROVED** [review]
the `SetByPath` method is complicated but it looks correct from first sight.
I guess we will add integration tests as well as unit tests, so we will have some more cases / some real cases covered.

File: internal/types/document.go [hunk 1]
```diff
@@ -311,8 +309,83 @@ func (d *Document) Remove(key string) any {
...
 
+// HasByPath returns true if the given path is present in the document.
+func (d *Document) HasByPath(path Path) bool {
+	_, err := d.GetByPath(path)
+
+	return err == nil
+}
+
+// GetByPath returns a value by path - a sequence of indexes and keys.
+// If the Path has only one element, it sets the value for the given key.
```
Comment (AlekSi) [inline internal/types/document.go:320]: That does not sound right

File: internal/types/document.go [hunk 2]
```diff
@@ -311,8 +309,83 @@ func (d *Document) Remove(key string) any {
...
+		}
+
+		must.NoError(inner.Set(index, value))
+	default:
+		panic(fmt.Errorf("can't set value for %T type", inner))
+	}
+}
+
 // RemoveByPath removes document by path, doing nothing if the key does not exist.
+// If the Path has only one element, it sets the value for the given key.
```
Comment (AlekSi) [inline internal/types/document.go:382]: Looks like incorrect copy&paste

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: internal/types/document.go (hunk 2 above)
Comment (w84thesun) [reply internal/types/document.go]: Yeah... Updated.

File: internal/types/document.go (hunk 1 above)
Comment (w84thesun) [reply internal/types/document.go]: Updated

**Review by AlekSi — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
LGTM

**Pull request merged**